site/docs/bazel-vision.md

```diff
@@ -0,0 +1,78 @@
+     the creative process of authoring code because the mechanical process of build and
+     test is solved. When customizing the build system to support new languages or unique
+     organizational needs, users focus on the aspects of extensibility that are unique to
+     their use case, without having to reinvent the basic plumbing.
+  *  **Engineers can easily contribute to any project.** A developer who wants to start 
+     working on a new project can simply clone the project and run the build. There’s no
+     need for local configuration - it just works. With cross-platform remote execution,
+     they can work on any machine anywhere and fully test their changes against all platforms
+     the project targets. Engineers can quickly configure the build for a new project or
+     incrementally migrate an existing build.
+  *  **Projects can scale to any size codebase, any size team.** Fast, incremental testing
+     allows teams to fully validate every change before it is committed. This remains true
+     even as repos grow, projects span multiple repos, and multiple languages are introduced.
+     Infrastructure does not force developers to trade test coverage for build speed.
+
+
+**We believe Bazel has the potential to fulfill this vision.** Bazel was built from the ground
+up to enable builds that are reproducible (a given set of inputs will always produce the same
+outputs) and portable (a build can be run on any machine without affecting the output). These
+characteristics support safe incrementality (rebuilding only changed inputs doesn't introduce
+the risk of corruption) and distributability (build actions are isolated and can be offloaded). 
+By minimizing the work needed to do a correct build and parallelizing that work across multiple
+cores and remote systems, Bazel can make any build fast. Bazel’s abstraction layer—instructions
+specific to languages, platforms, and toolchains implemented in a simple extensibility language
+— allows it to be easily applied to any context.
+
+## Bazel Core Competencies
+
+1. Bazel supports **multi-language, multi-platform** builds and tests. 
+   You can run a single command to build and test your entire source tree,
+   no matter which combination of languages and platforms you target.
+1. Bazel builds are **fast and correct**. Every build and test run is incremental,
+   on your developers' machines and on CI.
+1. Bazel provides a **uniform, extensible language** to define builds for any language
+   or platform.
+1. Bazel allows your builds **to scale** by connecting to remote execution and caching services.
+1. Bazel works across **all major development platforms** (Linux, MacOS, and Windows).
+1. We accept that adopting Bazel requires effort, but **gradual adoption** is possible.
+   Bazel interfaces with de-facto standard tools for a given language/platform.
+
+## Serving language communities
+
+Software engineering evolves in the context of language communities — typically,
```

Comment:
As someone applying Bazel to Apple based development, this really resonates! It seems like Bazel provides first class support for languages, and the idioms used by those communities. The pathological case is when the requirements/needs of the community conflict with what Bazel solves for and prioritizes. ( i.e. the `#include` style preferences of Bazel  https://github.com/bazelbuild/bazel/pull/3712 ), so it is good to see this addressed  🎉